### PR TITLE
add Scala 3 to crossbuild

### PR DESCRIPTION
**Pull request by SethTisue**
~includes commits from #393 — I'll rebase this one once that one's merged~

references #390

### CODE REVIEW

**Comment by SethTisue** [comment]
status: this is still in early stages, I am still just chewing through bread and have yet to take a bite of meat

File: build.sbt [hunk 1]
```diff
@@ -5,16 +5,16 @@ import org.scalajs.jsenv.nodejs.NodeJSEnv
 // Binary compatibility is this version
 val previousVersion: Option[String] = Some("1.5.0")
 
-val ScalaTestVersion              = "3.1.4"
-val ScalaXmlVersion               = "1.3.0"
-val ScalaParserCombinatorsVersion = "1.1.2"
+val ScalaTestVersion              = "3.2.3"
```
Comment (sullis) [inline build.sbt:8]: use ScalaTest 3.2.5

**Review by sullis — COMMENTED** [review]
ScalaTest 3.2.5

File: project/Common.scala [hunk 2]
```diff
@@ -28,9 +28,7 @@ object Common extends AutoPlugin {
     "-deprecation",
     "-feature",
     "-unchecked",
-    "-encoding",
-    "utf8",
-    "-target:jvm-1.8",
```
Comment (mkurz) [inline project/Common.scala:33]: Why did you remove `-target:jvm-1.8`?

Comment (SethTisue) [reply project/Common.scala:33]: It should have been removed at the time 2.11 support was dropped. It's the default on 2.12 and 2.13 and 3.0 regardless of what JVM you're building on.

**Comment by SethTisue** [comment]
Travis-CI config seems messed up, there is Scala 3 stuff in the Scala 2 jobs...

**Comment by SethTisue** [comment]
just a note that I haven't forgotten about this. I've been working on play-json instead, but this is next in line.

**Comment by SethTisue** [comment]
> Travis-CI config seems messed up, there is Scala 3 stuff in the Scala 2 jobs

Fixed now. But there are still two non-Scala 3 problems.

The plugin test job fails with `Conflicting cross-version suffixes`

And also, MiMa is failing:

```
[error] twirl-parser: Failed binary compatibility check against com.typesafe.play:twirl-parser_2.12:1.5.0! Found 1 potential problems
[error]  * method several(scala.Function0,scala.collection.mutable.Buffer,scala.reflect.Manifest)scala.collection.mutable.Buffer in class play.twirl.parser.TwirlParser's type is different in current version, where it is (scala.Function0,scala.collection.mutable.Buffer,scala.reflect.ClassTag)scala.collection.mutable.Buffer instead of (scala.Function0,scala.collection.mutable.Buffer,scala.reflect.Manifest)scala.collection.mutable.Buffer
[error]    filter with: ProblemFilters.exclude[IncompatibleMethTypeProblem]("play.twirl.parser.TwirlParser.several")
```

**Comment by SethTisue** [comment]
me up in `.travis.yml` trying to understand how `matrix:` and `stage:` interact:

tenor

**Comment by SethTisue** [comment]
Man, I hate builds that have an sbt plugin subproject mixed in with crossbuilt subprojects. It makes sense for such builds to exist, but 🙀

**Comment by SethTisue** [comment]
I think I got the CI matrix sorted out, including the plugin tests. I don't like omitting `plugin` from the root project aggregate, but I don't know how to improve that.

So what remains here is:

* MiMa is failing because `Manifest` vs `ClassTag`. I guess we ought to just bite the bullet here and get off `Manifest`, even if it means accepting the binary incompatibility and bumping the major version?
* the actual Scala 3 work

**Comment by SethTisue** [comment]
Still chipping away at this. A lot of PRs (mostly version bumps) were merged since I initially submitted this, so I rebased. And then I made separate `scala-2` and `scala-3` source directories with just the version-dependent bits in them, but right now the Scala 3 ones are just `???` stubs.

**Comment by SethTisue** [comment]
abandoning; someone else is free to pick it up

**Comment by SethTisue** [comment]
some of the context for this is at https://discuss.lightbend.com/t/github-project-abandonned/8781/12, by the way
